sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -121,7 +121,8 @@
         this.receiveMode = receiverOptions.getReceiveMode();
 
         this.entityType = entityType;
-        this.isSessionEnabled = isSessionEnabled;
+        this.isSessionEnabled = receiverOptions.isSessionEnabled();
```

Comment:
You don't need to create dedicated fields now. A reference to the receiverOptions is held by the class.